utils/publish-canary.js

```diff
@@ -11,28 +11,72 @@ const {
   TRAVIS_BUILD_URL = 'https://travis-ci.org/Workday/canvas-kit/branches',
 } = process.env;
 
+const isPrerelease = TRAVIS_BRANCH.match(/^prerelease\/v\d*$/g);
 const data = {};
-let preid;
+let distTag;
 
 if (TRAVIS_BRANCH === 'master') {
-  preid = 'next';
-} else if (TRAVIS_BRANCH.match(/^prerelease\/v\d*$/g)) {
-  preid = 'prerelease';
+  distTag = 'next';
+} else if (isPrerelease) {
+  // distTag = 'prerelease-next';
+  console.error('Prerelease canary builds disabled.');
+  process.exit(0);
 } else {
   console.error('No travis branch provided');
   process.exit(1);
 }
 
-cmd('git rev-parse --short HEAD')
+const slackAnnouncement = attachment => {
+  request.post(
+    SLACK_WEBHOOK,
+    {
+      json: {
+        attachments: [
+          {
+            ...attachment,
+            author_link: TRAVIS_BUILD_URL,
+            ts: Date.now(),
+          },
+        ],
+      },
+    },
+    (error, response, body) => {
+      if (error) {
+        throw error;
+      }
+    }
+  );
+};
+
+cmd('git diff --name-only HEAD HEAD^')
+  .then(filesChangedInMerge => {
+    if (filesChangedInMerge.includes('CHANGELOG.md')) {
+      console.log('Last merge commit was a release. Skipping canary build.');
+      process.exit(0);
+    }
+
+    return cmd('git rev-parse --short HEAD');
+  })
   .then(sha => {
     data.sha = sha.trim();
 
+    return cmd(`git describe --abbrev=0`);
+  })
+  .then(releaseTag => {
+    const lastReleasePreId = releaseTag.trim().split('-')[1];
+
+    const lastReleaseNum = parseInt(lastReleasePreId.match(/\d+$/), 10);
+    const pos = lastReleasePreId.indexOf(lastReleaseNum);
+    const nextReleasePreId = lastReleasePreId.slice(0, pos) + (lastReleaseNum + 1);
+
+    const preid = isPrerelease ? `${nextReleasePreId}-next` : 'next';
```

Comment:
Grabs last release tag (e.g. `4.0.0-beta.3`), and grabs the prerelease suffix (`beta.3`). It then grabs the number at the end, and increments it. Finally it appends `-next` (e.g. `beta.4-next`).

This preid will result in a version in the form of `4.0.0-beta.4-next.X` (where X is the integer number of commits since the `4.0.0-beta.3` release

Reply:
I think this can be done using the [semver package](https://www.npmjs.com/package/semver#prerelease-identifiers) too. You can probably get rid of a lot of the custom logic in case it has some bugs that we're not catching now.